migrations/9_create_user_emails_array.js

```diff
@@ -0,0 +1,49 @@
+const Settings = require('settings-sharelatex')
+const mongojs = require('mongojs')
+const db = mongojs(Settings.mongo.url, ['users'])
+const async = require('async')
+
+const handleExit = () => console.log('Got signal.  Shutting down.')
+process.on('SIGINT', handleExit)
+process.on('SIGHUP', handleExit)
+
+const initUserEmailsAttribute = (user, callback) => {
+  const update = {
+    $set: {
+      emails: [
+        {
+          email: user.email,
+          createdAt: new Date()
+        }
+      ]
+    }
+  }
+  db.users.update({ _id: user._id }, update, callback)
+}
+
+const updateAllUsersEmailsAttribute = (users, callback) => {
+  console.log(`updating ${user.length} users`)
+  async.eachSeries(users, initUserEmailsAttribute, callback)
```

Comment:
This could take a while -  might want to try `eachLimit` instead? There isn't any crosstalk between the user records here so it should be safe to do.

Reply:
I paired with @timothee-alby about this, we decided to get rid of it on ServerPro. It's a one-off migration performed on application startup, so that's not a problem.